# Upstream patch to fix crash if LADSPA_PATH is empty.
# Closes: #726197

diff -Naur mlt-0.9.0.orig/src/modules/jackrack/plugin_mgr.c mlt-0.9.0/src/modules/jackrack/plugin_mgr.c
--- mlt-0.9.0.orig/src/modules/jackrack/plugin_mgr.c	2013-06-03 05:34:35.000000000 +0200
+++ mlt-0.9.0/src/modules/jackrack/plugin_mgr.c	2013-10-17 14:49:57.147889337 +0200
@@ -237,10 +237,8 @@
     ladspa_path = g_strdup ("/usr/local/lib/ladspa:/usr/lib/ladspa:/usr/lib64/ladspa");
 #endif
   
-  dir = strtok (ladspa_path, ":");
-  do
+  for (dir = strtok (ladspa_path, ":"); dir; dir = strtok (NULL, ":"))
     plugin_mgr_get_dir_plugins (plugin_mgr, dir);
-  while ((dir = strtok (NULL, ":")));
 
   g_free (ladspa_path);
 }
